uwrt_utils/include/uwrt_utils/uwrt_can.h

```diff
@@ -0,0 +1,102 @@
+
+#include <linux/can.h>
+#include <linux/can/raw.h>
+#include <net/if.h>
+#include <chrono>
+#include <map>
+#include <mutex>
+#include <string>
+#include <thread>
+#include <vector>
+
+namespace uwrt_can {
+
+class UWRTCANWrapper {
+ public:
+  explicit UWRTCANWrapper() = default;
+  explicit UWRTCANWrapper(std::string name, std::string interface_name, int thread_sleep_millis = 1)
+      : name_(std::move(name)),
+        interface_name_(std::move(interface_name)),
+        thread_sleep_millis_(std::chrono::milliseconds(thread_sleep_millis)){};
+  // TODO (wraftus) make other constructor for failover CAN
+
+  explicit UWRTCANWrapper(const UWRTCANWrapper& to_copy) = delete;
+  explicit UWRTCANWrapper(UWRTCANWrapper&& to_move) noexcept
+      : name_(std::move(to_move.name_)),
+        interface_name_(std::move(to_move.interface_name_)),
+        thread_sleep_millis_(to_move.thread_sleep_millis_) {
+    to_move.read_thread_running_ = false;
```

Comment:
i think it makes more sense to move the thread and keep it executing rather than join it

Reply:
actually i see now why you do it because the init starts the processing. I don't think its right for the state of the class to change when you move or move construct it. maybe this means the init stuff should all be in the constructor and it should raise exceptions for errors instead of return values